src/json/emit-json.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+function emit({ content, response = {} }, { logger }) {
+  if (response.body) {
+    logger.debug('Response body already exists');
+    return {};
+  }
+  if (content && content.json) {
+    try {
+      logger.debug(`Emitting JSON from ${typeof content.json}`);
+      return {
+        response: {
+          body: JSON.stringify(content.json),
```

Comment:
I'm not sure about that. I thought OpenWhisk expects JSON responses as object literals. At least in the static action we check for text/binary/json and treat each one differently.